Client/src/test/functional/com/agilepro/client/helpers/TFBackLogHelper.java

```diff
@@ -0,0 +1,218 @@
+
+		/**
+		 * customer.
+		 **/
+		CustomerModel model = new CustomerModel("Test1", emailId, phoneNumber, null, null, null, null, new Date(), password, password, "path1", null);
+		model.setCustomerPricePlanId(idOfPricePlan);
+		model.setDueAmount(dueAmount);
+		customerId = customerHelper.save(clientContext, model);
+		Assert.assertTrue(idOfPricePlan > 0);
+		Assert.assertTrue(customerId > 0);
+		clientCurrentSession = super.newClientContext(emailId, password, customerId);
+		// camp type
+
+		clientControllerFactory = new ClientControllerFactory(clientCurrentSession);
+
+	}
+
+	/**
+	 * Gets the backlog.
+	 *
+	 * @param backlogId
+	 *            the backlog id
+	 * @return the backlog
+	 */
+	private BackLogModel getBacklog(long backlogId)
+	{
+		BasicReadResponse<BackLogModel> response = backlogcontroller.read(backlogId);
+		return response.getModel();
+	}
+
+	/**
+	 * Test save.
+	 */
+	@Test
+	public void testSave()
+	{
+		clientControllerFactory = new ClientControllerFactory(clientCurrentSession);
+
+		backlogcontroller = (IBackLogController) super.clientControllerFactory.getController(IBackLogController.class);
+
+		BackLogModel model = new BackLogModel();
+		model.setTitle("studentGrade");
+		model.setDescription("hello");
+		model.setEstimate(23);
+		BasicSaveResponse response2 = backlogcontroller.save(model);
+
+		Assert.assertEquals(response2.getCode(), IWebUtilsCommonConstants.RESPONSE_CODE_SUCCESS);
+		// validate save
+		getBacklog(response2.getId());
+	}
+
+	/**
+	 * Test update.
+	 */
+	@Test
+	public void testUpdate()
+	{
+		clientControllerFactory = new ClientControllerFactory(clientCurrentSession);
+
+		backlogcontroller = (IBackLogController) super.clientControllerFactory.getController(IBackLogController.class);
+
+		BackLogModel model = new BackLogModel();
+		model.setTitle("studentGrade");
+		model.setDescription("hello");
+		model.setEstimate(23);
+
+		// validate save
+		BasicSaveResponse response3 = backlogcontroller.save(model);
+
+		Assert.assertEquals(response3.getCode(), IWebUtilsCommonConstants.RESPONSE_CODE_SUCCESS);
+
+		// validate save
+		getBacklog(response3.getId());
+
+		BackLogModel model2 = new BackLogModel();
+		model2.setTitle("studentGraxxde");
+		model2.setDescription("hello");
+		model2.setEstimate(23);
+		model2.setId(response3.getId());
+		backlogcontroller.update(model2);
+	}
+
+	/**
+	 * Test delete.
+	 */
+	@Test
+	public void testDelete()
+	{
+		clientControllerFactory = new ClientControllerFactory(clientCurrentSession);
+
+		backlogcontroller = (IBackLogController) super.clientControllerFactory.getController(IBackLogController.class);
+
+		BackLogModel model = new BackLogModel();
+		model.setTitle("studentGrade");
+		model.setDescription("hello");
+		model.setEstimate(23);
+		BasicSaveResponse response = backlogcontroller.save(model);
+
+		Assert.assertEquals(response.getCode(), IWebUtilsCommonConstants.RESPONSE_CODE_SUCCESS);
+
+		// validate save
+		getBacklog(response.getId());

```

Comment:
Why to get backlog before delete. You should get it post delete and ensure u get null or ertror.